File: kata/bowling/roll.js
'use strict';

var Roll = (function() {
  function Roll(pins) {
    pins = pins || 0;

    if (isNaN(pins)) { throw new TypeError(); }
    if (pins < 0 || pins > 10) { throw new RangeError(); }

    this.pins = pins;
  }

  return Roll;
})();

module.exports.Roll = Roll;

File: katas/bowlingJava/src/bowlingKata/Roll.java
package bowlingKata;

public class Roll
{
	public int pins;

	public Roll()
	{
		this.pins = 0;
	}
	
	public Roll(int pins)
	{
		this.pins = pins;
	}
	
	public int nbPins()
	{
		return pins;
	}
	
	public boolean allPinsKnockedDown()
	{
		return pins == 10;
	}
}